beater/beater.go

```diff
@@ -136,10 +114,42 @@ type beater struct {
 	stopped    bool
 }
 
+var once sync.Once
+
 // Run runs the APM Server, blocking until the beater's Stop method is called,
 // or a fatal error occurs.
 func (bt *beater) Run(b *beat.Beat) error {
 
+	done := make(chan struct{})
+
+	var reloadable = func() reload.Reloadable {
+		return reload.ReloadableFunc(func(ucfg *reload.ConfigWithMeta) error {
```

Comment:
```suggestion
	reloadable := reload.ReloadableFunc(func(ucfg *reload.ConfigWithMeta) error {
```

Then you can just use it directly in Register, i.e. `reload.Register.MustRegister("inputs", reloadable)`